File: Python/1764.py
import sys

f = lambda : sys.stdin.readline().rstrip()

line = list(map(int, f().split()))

N = line[0]
M = line[1]

dic = dict()
for i in range(N):
    name = f()
    dic[name] = "듣"

result = list()
for i in range(M):
    name = f()
    if name in dic.keys():
        result.append(name)

print(len(result))
for i in sorted(result):
    print(i)

File: Java/_1764.java
package Baekjoon;

import java.util.*;
import java.io.*;

public class _1764 {

    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] line = br.readLine().split(" ");

        int N = Integer.parseInt(line[0]);
        int M = Integer.parseInt(line[1]);

        Map<String, String> map = new HashMap<>();

        for(int i = 0; i < N; i++){
            String name = br.readLine();
            map.put(name, "듣");
        }

        List<String> result = new ArrayList<>();

        for(int i = 0; i < M; i++){
            String name = br.readLine();
            if(map.containsKey(name)){
                result.add(name);
            }
        }

        Collections.sort(result);

        System.out.println(result.size());
        for(String i : result){
            System.out.println(i);
        }
    }
}
